pkg/resources/kafka/kafka.go

```diff
@@ -804,3 +798,25 @@ func GetBrokersWithPendingOrRunningCCTask(kafkaCluster *v1beta1.KafkaCluster) []
 func isDesiredStorageValueInvalid(desired, current *corev1.PersistentVolumeClaim) bool {
 	return desired.Spec.Resources.Requests.Storage().Value() < current.Spec.Resources.Requests.Storage().Value()
 }
+
+func (r *Reconciler) createExternalListenerStatuses() (map[string]v1beta1.ListenerStatus, error) {
+	extListenerStatuses := make(map[string]v1beta1.ListenerStatus, len(r.KafkaCluster.Spec.ListenersConfig.ExternalListeners))
+	for _, eListener := range r.KafkaCluster.Spec.ListenersConfig.ExternalListeners {
```

Comment:
Doesn't cover the case when `NodePort` is used to expose the Kafka cluster. For examples see:
https://github.com/banzaicloud/kafka-operator/blob/master/config/samples/simplekafkacluster-with-nodeport-external.yaml#L36

https://github.com/banzaicloud/kafka-operator/blob/master/config/samples/simplekafkacluster-with-nodeport-external.yaml#L55